src/com/pensio/api/PensioMerchantAPI.java

```diff
@@ -288,6 +297,21 @@ protected void setPaymentRequestParameters(
 		addPaymentInfo(paymentRequest, params);
 
 		addOrderLines("orderLines",params, paymentRequest.getOrderLines());
+
+		if(paymentRequest.getAgreementConfig() != null)
+		{
+			addParam(params, "agreement[id]", paymentRequest.getAgreementConfig().getAgreementId());
+			addParam(params, "agreement[type]", paymentRequest.getAgreementConfig().getAgreementType().name());
+			addParam(params, "agreement[unscheduled_type]", paymentRequest.getAgreementConfig().getAgreementUnscheduledType().name());
+			addParam(params, "agreement[expiry]", DateHelper.formatDate("yyyyMMdd", paymentRequest.getAgreementConfig().getAgreementExpiry()));
+			addParam(params, "agreement[frequency]", paymentRequest.getAgreementConfig().getAgreementFrequency());
+			addParam(params, "agreement[next_charge_date]", DateHelper.formatDate("yyyyMMdd", paymentRequest.getAgreementConfig().getAgreementNextChargeDate()));
+			addParam(params, "agreement[admin_url]", paymentRequest.getAgreementConfig().getAgreementAdminUrl());
+		}
+		//below check is to be removed as getAgreementType() is deprecated.
+		if(paymentRequest.getAgreementType() != null){
```

Comment:
if `getAgreementType()` is already deprecated, why do we use it here? why not use an alternative, that is not deprecated?

this kind of "todo" comments to remove deprecated calls are useful for the code that already exists. but writing new code, that uses deprecated methods, should be avoided

Reply:
This code already existed in `addAgreementType()`. I removed this function and placed the code here.